src/token_contract/src/main.nr

```diff
@@ -630,14 +608,6 @@ pub contract Token {
             from,
         );
 
-        // We can't simply return the partial note because we won't be able to later on verify that it was created
```

Comment:
what if we do exactly that? simply return the partial note, it's being emitted inside the `.partial` method, encrypted to the recipient, so what's the point of storing the commitment in public storage?

this doesn't just save up l2 gas, but also makes txs `transfer_to_public{/_with_commitment}` fungible!